### PR TITLE
lf em 4x70 autorecover

### PR DESCRIPTION
**Pull request by henrygab**
Add `lf em 4x70 autorecover` command

Also:
* common.h: Increase safety of some of the macros.
  Parenthesizing the macro parameters ... a best practice.
* firmware: partially-tracked source of "extra bits" messages.
  Add a TODO comment for further study.
* Improve reliability of `lf em 4x70 writekey`
  Authenticate w/new key after it is written.
  Particularly important for glass modules,
  or other tags with weaker coupling.

### CODE REVIEW

File: client/src/cmdlfem4x70.c [hunk 1]
```diff
@@ -774,21 +1037,365 @@
...
+                .partial_key_start = {0},
+            };
+
+            result = brute_em4x70(&opts_brute, &brute);
+            if (PM3_ETIMEOUT == result) {
+                PrintAndLogEx(FAILED, "Timeout while waiting for reply.");
+                PrintAndLogEx(HINT, "Block %d data was overwritten.  Manually restart at step %d.", block, step);
+                return result;
+            } else if (PM3_SUCCESS != result) {
+                PrintAndLogEx(FAILED, "Writing block %d: " _RED_("failed") ".");
```
[the remark this answers is not in the document]
Comment (henrygab) [reply client/src/cmdlfem4x70.c]: fixed w/7fbc6b2

**Comment by iceman1001** [comment]
Nice one,

I see an issue with your setup doesn't have GD available and hence influence the `make style`by removing waveshare.

Consider using  `setpin` and `setkey`  instead of `writepin`,  `writekey` ,

**Comment by henrygab** [comment]
> I see an issue with your setup doesn't have GD available and hence influence the `make style`by removing waveshare.

Oh!  That's unexpected!  I didn't realize `make style` (at least for the documentation portions) would exclude options that are not built.

1. Is the documentation portion of `make style` something that can be automatically / periodically done within a github action?   (to ensure it's always up-to-date, for example)

2. I've submitted PR #2328 as a manual fix.

3. I chose not to update the `ID48LIB` library sources manually ... is there a way to exclude that from `make style` altogether so others don't mess with it?

> Consider using `setpin` and `setkey` instead of `writepin`, `writekey` ,

4. Just want to confirm ... are you asking me to modify the verbs?  Even if breaking existing scripts?

I don't mind doing the work to do so, and so I'd be happy to do it.  I just wanted to ensure you'd provide "air cover" if someone gets upset about my breaking their scripts.

Thanks!

**Comment by iceman1001** [comment]
I don't think we thought the the change to GD dependency would be not installed by default on a OS. 

If I knew,  I wouldn't have approved the change.

You would need to look the `make style` commands in the makefile and see if you can make an exception for your lib.

Which existing scripts uses `writepin`  or `writekey` ?    When I search the repo there is not any hits in a script.

**Comment by henrygab** [comment]
> Which existing scripts uses `writepin` or `writekey` ? When I search the repo there is not any hits in a script.

None that are checked in.  Let me know either way, I'm happy to adjust to your preferences.

**Comment by iceman1001** [comment]
Since we have no issue with repo,   please adjust to the suggested new command names

**Pull request merged**